src/library/scala/collection/StringOps.scala

```diff
@@ -159,8 +159,20 @@ final class StringOps(private val s: String)
     *  @return a new string which is a copy of this string with the element at position `index` replaced by `elem`.
     *  @throws IndexOutOfBoundsException if `index` does not satisfy `0 <= index < length`.
     */
-  def updated(index: Int, elem: Char): String =
-    fromSpecificIterable(new View.Updated(this, index, elem)) // TODO optimize
+  def updated(index: Int, elem: Char): String = {
+    val sb = new java.lang.StringBuilder(s)
+    sb.setCharAt(index, elem)
+    sb.toString()
+  }
+
+  // override for performance
+  /** Tests whether this $coll contains the given character.
+   *
+   *  @param elem  the character to test.
+   *  @return     `true` if this $coll has an element that is equal (as
+   *              determined by `==`) to `elem`, `false` otherwise.
+   */
+  override def contains[C >: Char](elem: C): Boolean = s.indexOf(elem) >= 0
```

Comment:
I would expect this to be an overload, rather than an override.

```
override def contains(elem: Char): Boolean = s.indexOf(elem) >= 0
```

@szeiger WDYT?

Reply:
Hmmm, perhaps I've confused myself here. This looks good.

Reply:
But it needs to be:

```scala
  override def contains[C >: Char](elem: C): Boolean = elem match {
    case c: Char => s.indexOf(c) >= 0
    case _ => super.contains(elem)
  }

```

Reply:
I'm confused as well. char equality means that nothing but a char will return true, right?

Reply:
I think you get the enriched `StringOps.indexOf` with generic `C`? not native string indexOf.

Reply:
```
override def contains[C >: Char](elem: C): Boolean = elem match {
    case c: Char => s.indexOf(c) >= 0
    case _ => false
  }
```
could you do this?

Reply:
Probably. The old machinery uses apply/charAt and == so there's no way to stumble across an element that redefines equals. I haven't followed new collections yet.

Reply:
That looks like it would change `new StringOps("a").contains('a'.toInt)` to false. Currently it is true, due to cooperative equality:

```
scala> Int.box(1) == Char.box(1)
res1: Boolean = true

scala> "a".contains('a'.toInt)
res3: Boolean = true
```